tests/drivers/rtio/rtio_api/src/test_rtio.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <rtio.h>
+
+RTIO_POOL_DEFINE(small_pool, 1024, 1);
+
+void test_rtio_pool(void)
+{
+	struct rtio_block *block;
+	struct rtio_block *block2;
+
+	u32_t size = 1024 - sizeof(struct rtio_block);
+	int res = rtio_block_alloc(&small_pool, &block, size, K_NO_WAIT);
+
+	zassert_equal(res, 0, "Allocation should not fail");
+	zassert_equal(block->size, size, "Size should be 1024");
```

Comment:
minus sizeof(struct rtio_block), no?

Reply:
This should be corrected now, as the allocator API accounts for any metadata sizing